src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs

```diff
@@ -53,63 +122,170 @@ public void TestGetAndListVirtualNetworkRule()
-            }, SqlManagementTestUtilities.DefaultEuapPrimaryLocation);
+
+            }
         }
 
         [Fact]
         public void TestCreateAndDropVirtualNetworkRule()
         {
-            // TODO Update SQL VirtualNetworkRules test to be fully automated
-            // TODO https://github.com/Azure/azure-sdk-for-net/issues/3388
 
             string testPrefix = "virtualnetworkrulecrudtest-";
-            string suiteName = this.GetType().FullName;
-            SqlManagementTestUtilities.RunTestInNewV12Server(suiteName, "TestCreateAndDropVirtualNetworkRule", testPrefix, (resClient, sqlClient, resourceGroup, server) =>
+
+            using (SqlManagementTestContext context = new SqlManagementTestContext(this))
             {
-                // Create and validate Firewall Rule
-                //
+                // Basic setup
+                var location = SqlManagementTestUtilities.DefaultEuapPrimaryLocationId;
+                ResourceGroup resourceGroup = context.CreateResourceGroup(location);
+                NetworkManagementClient networkClient = context.GetClient<NetworkManagementClient>();
+
+                // Create vnet 
+                string vnetName = SqlManagementTestUtilities.GenerateName(testPrefix);
+                string subnetName = SqlManagementTestUtilities.GenerateName(testPrefix);
+
+                List<PrivateAccessServicePropertiesFormat> SqlPrivateAccess = new List<PrivateAccessServicePropertiesFormat>();
+                SqlPrivateAccess.Add(new PrivateAccessServicePropertiesFormat("Microsoft.Sql"));
+
+                var vnet = new VirtualNetwork()
+                {
+                    Location = location,
+
+                    AddressSpace = new AddressSpace()
+                    {
+                        AddressPrefixes = new List<string>()
+                        {
+                            "10.0.0.0/16",
+                        }
+                    },
+                    DhcpOptions = new DhcpOptions()
+                    {
+                        DnsServers = new List<string>()
+                        {
+                            "10.1.1.1",
+                            "10.1.2.4"
+                        }
+                    },
+                    Subnets = new List<Subnet>()
+                    {
+                        new Subnet()
+                        {
+                            Name = subnetName,
+                            AddressPrefix = "10.0.1.0/24",
+                            PrivateAccessServices = SqlPrivateAccess,
+                        }
+                    }
+                };
+
+                // Put Vnet
+                var putVnetResponse = networkClient.VirtualNetworks.CreateOrUpdate(resourceGroup.Name, vnetName, vnet);
+                Assert.Equal("Succeeded", putVnetResponse.ProvisioningState);
+
+                // Get Vnet
+                var getVnetResponse = networkClient.VirtualNetworks.Get(resourceGroup.Name, vnetName);
```

Comment:
all of the above is the same for each test, right? please extract common code to a method, e.g. `VirtualNetwork vnet = CreateVirtualNetwork(context, location);`